src/policy/immix/immixspace.rs

```diff
@@ -68,9 +68,57 @@ impl<VM: VMBinding> SFT for ImmixSpace<VM> {
             self.is_marked(object, self.mark_state) || ForwardingWord::is_forwarded::<VM>(object)
         }
     }
+    fn pin_object(&self, object: ObjectReference) -> bool {
```

Comment:
It seems the pinning here is general, and there is nothing specific with Immix. In that case you can just implement `pin/unpin/is_pinned` on `VMLocalPinningBitSpec`, like what we did for the log bit.

https://github.com/mmtk/mmtk-core/blob/644628bcc39e649e4125c092ad325bddbd24aaaf/src/util/metadata/log_bit.rs#L6-L11